Add ESLint rule for throwing only error objects (#892)
goldbergyoni/nodebestpractices: README.md

## ![✔] 2.2 Use only the built-in Error object

> **TL;DR:** Many throw errors as a string or as some custom type – this complicates the error handling logic and the interoperability between modules. Whether you reject a promise, throw an exception or emit an error – using only the built-in Error object (or an object that extends the built-in Error object) will increase uniformity and prevent loss of information. There is `no-throw-literal` ESLint rule that strictly checks that (although it have some [limitations](https://eslint.org/docs/rules/no-throw-literal) which can be solved when using TypeScript and setting the `@typescript-eslint/no-throw-literal` rule)

Reviewer: ```suggestion
**TL;DR:** Many throw errors as a string or as some custom type – this complicates the error handling logic and the interoperability between modules. Whether you reject a promise, throw an exception or emit an error – using only the built-in Error object (or an object that extends the built-in Error object) will increase uniformity and prevent loss of information.

There is a `no-throw-literal` ESLint rule that strictly checks that (although it has some [limitations](https://eslint.org/docs/rules/no-throw-literal), which can be solved when using TypeScript and using the `@typescript-eslint/no-throw-literal` rule).
```

Reviewer 2: Thank you @elektronik2k5 for your suggestion however we try to keep a two block per bullet point "style". One `tl;dr` and one `otherwise`. Sometimes we also add examples to those two but that's it, no extra linebreaks.

**TL;DR:** Many throw errors as a string or as some custom type – this complicates the error handling logic and the interoperability between modules. Whether you reject a promise, throw an exception or emit an error – using only the built-in Error object (or an object that extends the built-in Error object) will increase uniformity and prevent loss of information

> There is `no-throw-literal` ESLint rule that strict that (although it have some [limitations](https://eslint.org/docs/rules/no-throw-literal))

Reviewer 2: I think we're missing a `check` 🙂 : `that strictly checks that`

Reviewer 2: Also, let's add you sentence right after `loss of information.` We tend to avoid line breaks on the main page
